Sources/SwiftInspectorCore/PropertyAnalyzer.swift

```diff
@@ -0,0 +1,324 @@
+      return result.union(modifier)
+    }
+
+    // If there are no explicit modifiers, this is an internal property
+    if !modifier.contains(.public) &&
+      !modifier.contains(.fileprivate) &&
+      !modifier.contains(.private)
+    {
+      modifier = modifier.union(.internal)
+    }
+
+    // If the variable isn't static, it's an instance variable
+    if !modifier.contains(.static) {
+      modifier = modifier.union(.instance)
+    }
+
+    return modifier
+  }
+}
+
+// MARK: - TypeProperties
+
+/// Information about a type as well as its associated properties
+public struct TypeProperties: Hashable {
+
+  public struct Modifier: Hashable, OptionSet {
+    public let rawValue: Int
+
+    public static let `internal` = Modifier(rawValue: 1 << 0)
+    public static let `public` = Modifier(rawValue: 1 << 1)
+    public static let `private` = Modifier(rawValue: 1 << 2)
+    public static let privateSet = Modifier(rawValue: 1 << 3)
+    public static let `fileprivate` = Modifier(rawValue: 1 << 4)
+    public static let `instance` = Modifier(rawValue: 1 << 5)
+    public static let `static` = Modifier(rawValue: 1 << 6)
+
+    public init(rawValue: Int)  {
+      self.rawValue = rawValue
+    }
+
+    public init(stringValue: String) {
+      switch stringValue {
+      case "public": self = .public
+      case "private": self = .private
+      case "private(set)": self = .privateSet
+      case "fileprivate": self = .fileprivate
+      case "internal": self = .internal
+      case "static": self = .static
+      default: self = []
+      }
+    }
+  }
+
+  public struct PropertyData: Hashable {
+    /// The name of the property
+    public let name: String
+    /// The Type annotation of the property if it's present
+    public let typeAnnotation: String?
+    /// Any comments associated with the property
+    public let comment: String
+    /// Modifier set for this type
+    public let modifiers: Modifier
+  }
+
+  /// The name of the type.
+  public let name: String
+  /// The properties on this type
+  public let properties: [PropertyData]
+}
+
+extension TypeProperties {
+  struct MergeError: Error, CustomStringConvertible {
+
+    init(errorMessage: String) {
+      self.errorMessage = errorMessage
+    }
+
+    var description: String { errorMessage }
+
+    private let errorMessage: String
+  }
+}
+
+extension TypeProperties.MergeError {
+  static var invalidNames = TypeProperties.MergeError(
+    errorMessage: "Invalid types - the type names must match to be merged.")
+}
+
+extension TypeProperties {
+  func merge(with other: TypeProperties?) throws -> TypeProperties {
+    guard let other = other else {
+      return self
+    }
+    guard name == other.name else {
+      throw MergeError.invalidNames
+    }
+    return TypeProperties(
+      name: name,
+      properties: Array(Set(properties).union(Set(other.properties))))
+  }
+}
+
+extension TypeProperties.Modifier: CustomStringConvertible, CustomDebugStringConvertible {
```

Comment:
Let's move this extension to the `Command` since this is related to how we output things to the standard output which the Analyzer shouldn't care about see: https://github.com/fdiaz/SwiftInspector/blob/main/CONTRIBUTING.md#swiftinspectorcommand